packages/backend-app-api/src/services/implementations/auth/PluginTokenHandler.ts

```diff
@@ -114,21 +116,25 @@ export class PluginTokenHandler {
       throw new AuthenticationError('Invalid plugin token', e);
     });
 
-    return { subject: payload.sub };
+    return { subject: `plugin:${payload.sub}`, limitedUserToken: payload.obo };
   }
 
   async issueToken(options: {
     pluginId: string;
     targetPluginId: string;
+    onBehalfOf?: { token: string; expiresAt: Date };
   }): Promise<{ token: string }> {
+    const { pluginId, targetPluginId, onBehalfOf } = options;
     const key = await this.getKey();
 
-    const sub = options.pluginId;
-    const aud = options.targetPluginId;
+    const sub = pluginId;
+    const aud = targetPluginId;
     const iat = Math.floor(Date.now() / 1000);
-    const exp = iat + this.keyDurationSeconds;
+    const exp = onBehalfOf
+      ? Math.floor(onBehalfOf.expiresAt.getTime() / 1000)
+      : iat + this.keyDurationSeconds;
```

Comment:
as mentioned elsewhere - the expiry code needs fixin' so that tokens issued near expiration don't fail

Reply:
Fix in https://github.com/backstage/backstage/pull/24077

Reply:
Pushed https://github.com/backstage/backstage/pull/24024/commits/7978a58ded5fd0bc7e819abb3a408dadc7cbb866 for the fix here, @vinzscam working on tests async so let's make sure we cover this as well